### PR TITLE
ISSUES-863 fix test failure 

### PR DESCRIPTION
**Pull request by zhang2014**
#2015 #863
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

### CODE REVIEW

**Comment by zhang2014** [comment]
details of error: https://travis-ci.org/yandex/ClickHouse/jobs/416204943#L1889

File: dbms/tests/queries/0_stateless/00687_top_and_offset.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+${CLICKHOUSE_CLIENT} --query "INSERT INTO test.test VALUES (6);"
+${CLICKHOUSE_CLIENT} --query "INSERT INTO test.test VALUES (7);"
+${CLICKHOUSE_CLIENT} --query "INSERT INTO test.test VALUES (8);"
+${CLICKHOUSE_CLIENT} --query "INSERT INTO test.test VALUES (9);"
+
+${CLICKHOUSE_CLIENT} --query "SELECT TOP 2 * FROM test.test ORDER BY val;"
+${CLICKHOUSE_CLIENT} --query "SELECT TOP (2) * FROM test.test ORDER BY val;"
+${CLICKHOUSE_CLIENT} --query "SELECT * FROM test.test ORDER BY val LIMIT 2 OFFSET 2;"
+
+echo `${CLICKHOUSE_CLIENT} --query "SELECT TOP 2 * FROM test.test ORDER BY val LIMIT 2;" 2>&1 | grep -c "Code: 406"`
```
Comment (zhang2014) [inline dbms/tests/queries/0_stateless/00687_top_and_offset.sh]: I've made a few changes here，Please review it for me : ) @VadimPE

Comment (alexey-milovidov) [reply dbms/tests/queries/0_stateless/00687_top_and_offset.sh]: Ok, but
```
-- { clientError 406 }
```
annotations should work.
Look at 
https://github.com/yandex/ClickHouse/pull/2744
https://github.com/yandex/ClickHouse/pull/2761

It requires more investigation.

Comment (zhang2014) [reply dbms/tests/queries/0_stateless/00687_top_and_offset.sh]: Ok, I can keep track of this.

**Pull request merged**